### PR TITLE
WebBrowser embedded webview on NET5 (parity with NET fwk)

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+using System.Net.NetworkInformation;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.UI;
+
+namespace Microsoft.Identity.Client.Platforms.net45
+{
+    internal class WebView2WebUI : IWebUI
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs:14]: nit: is there a better name for it?

File: src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                {
+                    // Need to catch the exception here and put on the TCS which is the task we are waiting on so that
+                    // the exception comming out of Authenticate is correctly thrown.
+                    ((TaskCompletionSource<object>)tcs).TrySetException(e);
+               }
+            });
+
+            // If the thread is MTA, it cannot create or communicate with WebBrowser which is a COM control.
+            // In this case, we have to create the browser in an STA thread via StaTaskScheduler object.
+            //if (Thread.CurrentThread.GetApartmentState() == ApartmentState.MTA)
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs:53]: should this be commented?

File: src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            //    sendAuthorizeRequest();
+            //}
+
+            sendAuthorizeRequest();
+
+            return await Task.Factory.StartNew(() => authorizationResult).ConfigureAwait(false);
+        }
+
+        //internal AuthorizationResult Authenticate(Uri requestUri, Uri callbackUri)
+        //{
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs:102]: will something happen here?

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Comment by jennyf19** [comment]
looks good, just some questions in the webUI class.

File: src/client/Microsoft.Identity.Client/Platforms/net5win/Net5WebUiFactory.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Platforms/net5win/Net5WebUiFactory.cs:1]:
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

﻿using System;
```

File: src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs (hunk 3 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs:102]: I'll get that part updated or I'll remove it completely.

File: tests/devapps/WAM/NetCoreWinFormsWam/Form1.cs [hunk 5]
```diff
@@ -295,46 +317,48 @@ private bool IsMsaPassthroughConfigured()
             return msa;
         }
 
-        private Prompt GetPrompt()
+        private Prompt? GetPrompt()
```
Comment (neha-bhargava) [inline tests/devapps/WAM/NetCoreWinFormsWam/Form1.cs:320]: This is interesting. What does it imply? Return value is optional?

**Review by neha-bhargava — APPROVED** [review]

**Comment by jmprieur** [comment]
does bogavril/webview2 build for you?

**Comment by bgavrilMS** [comment]
No, this is work in progress. WebBrowser works for me and I think I will commit this as a standalone improvement.

File: tests/devapps/WAM/NetCoreWinFormsWam/Form1.cs (hunk 5 above)
Comment (bgavrilMS) [reply tests/devapps/WAM/NetCoreWinFormsWam/Form1.cs:320]: Nullable values. In C#, as in Java, value types (int, bool, enums, structs like Prompt) cannot be null. C# added the concept of "nullable value", where you add a `?` after the type, e.g. `int?` means nullable int etc. You can then do things like: 

```csharp
int? i = null; 
int? j = 14;

i.HasValue // false
j.HasValue // true
j.Value // 14
i.Value // exception I think

if (i == 14) { .. } // is equivalent to 
if (i.HasValue && i.Value ==14) { ...} 

For more details see https://docs.microsoft.com/en-us/dotnet/api/system.nullable-1.value?view=net-5.0

```

File: src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsWebUi/WebView2/WebUI.cs]: I'll repurpose this for only WebBrowser

**Pull request merged**